### DISCUSSION TITLE
Trading pallet

### DISCUSSION THREAD

**Discussion by Leouarz** [capsule-corp-ternoa/ternoa-proposals#44] [Ideas]
Starting a discussion about a new pallet used for trending NFTs with other NFTs and / or caps.

Simple implementation : 
- I want to trade my NFTs with someone else's NFT with some caps
- I create a trade contract that can be accepted or declined
- A trade must contain at least one NFT
- A trade can be agains NFTs and / or Caps (same as Monopoly game)

Advanced implementation : 
- I lock my NFT(s) for a trade
- I wait for offers (can be NFTs and / or Caps)
- I can chose among multiple offers to get the best

This will allow : 
- Safe Trading of NFTs
- Reserving an NFTs for someone (without having to put it on sale since everybody can buy it)
- Some kind of marketplace where currency is NFTs and / or Caps and not Caps only
- Maybe even a system where NFTs can receive offers directly without a trade creation beforehand

Will be used : 
- P2P
- Games
- Marketplaces

Don't hesitate to comment on this 🌸

**Comment by peshwar9** [comment]
What would a trade contract look like for swapping NFTs?

**Reply by Leouarz** [reply]
It would look like this : 

```rust
pub struct DealData {
  offerer: AccountId, 
  offeree: AccountId,
  offererSide: DealSide,
  offereeSide: DealSide,
}

pub struct DealSide {
  NFTs: Option<NFTList>,
  Amount: Option<Balance>,
}
```

With the constraint that at least 1 DealSide has at least NFT

**Comment by peshwar9** [comment]
@Leouarz : It could be useful to write a TIP (Ternoa Improvement Proposal) for this feature you are proposing.

**Reply by Leouarz** [reply]
Noice
